go/vt/wrangler/traffic_switcher.go

```diff
@@ -859,8 +859,45 @@ func (wr *Wrangler) DropSources(ctx context.Context, targetKeyspace, workflowNam
 	return sw.logs(), nil
 }
 
+func (wr *Wrangler) getShardSubset(ctx context.Context, keyspace string, shardSubset []string) ([]string, error) {
+	if wr.WorkflowParams != nil && len(wr.WorkflowParams.ShardSubset) > 0 {
+		shardSubset = wr.WorkflowParams.ShardSubset
+	}
+	allShards, err := wr.ts.GetShardNames(ctx, keyspace)
+	if err != nil {
+		return nil, err
+	}
+	if len(allShards) == 0 {
+		return nil, fmt.Errorf("no shards found in keyspace %s", keyspace)
+	}
+
+	if len(shardSubset) == 0 {
+		return allShards, nil
+	} else {
```

Comment:
Again, we can skip the `else` clause and unindent the reminder of this function.